mozilla_django_oidc_db/migrations/0007_migrate_oidc_data.py

```diff
@@ -0,0 +1,144 @@
+    )
+    old_config.oidc_op_token_endpoint = (
+        new_config.oidc_provider_config.oidc_op_token_endpoint
+    )
+    old_config.oidc_op_user_endpoint = (
+        new_config.oidc_provider_config.oidc_op_user_endpoint
+    )
+    old_config.oidc_op_logout_endpoint = (
+        new_config.oidc_provider_config.oidc_op_logout_endpoint
+    )
+
+    old_config.oidc_token_use_basic_auth = new_config.oidc_token_use_basic_auth
+    old_config.oidc_rp_idp_sign_key = new_config.oidc_rp_idp_sign_key
+    old_config.oidc_use_nonce = new_config.oidc_use_nonce
+    old_config.oidc_nonce_size = new_config.oidc_nonce_size
+    old_config.oidc_state_size = new_config.oidc_state_size
+    old_config.oidc_keycloak_idp_hint = new_config.oidc_keycloak_idp_hint
+    old_config.userinfo_claims_source = new_config.userinfo_claims_source
+
+    old_config.username_claim = new_config.options["user_claim_mappings"]["username"]
+
+    other_user_mappings = new_config.options["user_claim_mappings"]
+    del other_user_mappings["username"]
+
+    old_config.claim_mapping = other_user_mappings
+
+    group_settings = new_config.options.get("groups_settings", {})
+    old_config.groups_claim = group_settings.get("claim_mapping", [])
+    old_config.sync_groups = group_settings.get("sync", True)
+    old_config.sync_groups_glob_pattern = group_settings.get("sync_pattern", "*")
+    old_config.make_users_staff = group_settings.get("make_users_staff", False)
+    old_config.superuser_group_names = group_settings.get("superuser_group_names", [])
+    old_config.save()
+
+    groups = [
+        Group.objects.get_or_create(name=group_name)[0]
+        for group_name in group_settings.get("default_groups", [])
+    ]
+
+    old_config.default_groups.set(groups)
+    old_config.save()
+
+
+def move_data_forward(apps, schema_editor):
+    OIDCConfig = apps.get_model("mozilla_django_oidc_db", "OIDCConfig")
+    OIDCProviderConfig = apps.get_model("mozilla_django_oidc_db", "OIDCProviderConfig")
+    OpenIDConnectConfig = apps.get_model(
+        "mozilla_django_oidc_db", "OpenIDConnectConfig"
+    )
+
+    # Only one solo model should be present
+    old_config = OpenIDConnectConfig.objects.first()
+    if not old_config:
+        return
+
+    oidc_provider_config = OIDCProviderConfig()
+    oidc_provider_config.identifier = "admin-oidc-provider"
+    oidc_provider_config.oidc_op_discovery_endpoint = (
+        old_config.oidc_op_discovery_endpoint
+    )
+    oidc_provider_config.oidc_op_jwks_endpoint = old_config.oidc_op_jwks_endpoint
+    oidc_provider_config.oidc_op_authorization_endpoint = (
+        old_config.oidc_op_authorization_endpoint
+    )
+    oidc_provider_config.oidc_op_token_endpoint = old_config.oidc_op_token_endpoint
+    oidc_provider_config.oidc_op_user_endpoint = old_config.oidc_op_user_endpoint
+    oidc_provider_config.oidc_op_logout_endpoint = old_config.oidc_op_logout_endpoint
+    oidc_provider_config.save()
```

Comment:
is there a reason why you don't use `OIDCProviderConfig.objects.create(...kwargs)`?